quickcomm/admin.py

```diff
@@ -1,13 +1,337 @@
+
+        self.message_user(request, "Likes cleared successfully")
+        return HttpResponseRedirect(".")
+
+    @admin.action(description='Sync Comments')
+    def sync_comments(self, request, queryset):
+        for post in queryset:
+            if post.author.is_local:
+                continue
+            sync_comments(post)
+
+        self.message_user(request, "Comments synced successfully")
+        return HttpResponseRedirect(".")
+
+    @admin.action(description='Clear Comments')
+    def clear_comments(self, request, queryset):
+        for post in queryset:
+            if post.author.is_local:
+                continue
+            post.comments.all().delete()
+
+        self.message_user(request, "Comments cleared successfully")
+        return HttpResponseRedirect(".")
+@admin.register(HostAuthenticator)
+class HostAuthenticatorAdmin(admin.ModelAdmin):
+    list_display = ('username', 'nickname')
+    readonly_fields = ('base64',)
+    actions_on_top = True
+
+    @admin.display(description='Base64 Encoded Username:Password')
+    def base64(self, obj: HostAuthenticator):
+        return obj.base64string
+class PostInlineForAuthor(admin.TabularInline):
+    model = Post
+    extra = 0
+    fields = ['title', 'description']
+    show_change_link = True
+    can_delete = False
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    @admin.display(description='Title')
+    def title(self, obj: Post):
+        return obj.title
+
+
+@admin.register(Author)
+class AuthorAdmin(admin.ModelAdmin):
+    list_display = ('display_name', 'host', 'location')
+    readonly_fields = ('host', 'external_url', 'location', 'profile_image_html')
+    actions_on_top = True
+    actions = ['sync_posts', 'clear_posts', 'sync_followers', 'clear_followers']
+    inlines = [PostInlineForAuthor]
+
+    change_form_template = 'admin/edit_author.html'
+
+    fieldsets = (
+        (None, {
+            'fields': ('display_name', 'github', 'profile_image', 'profile_image_html', 'location', 'user')
+        }),
+        ('Host Details', {
+            'fields': ('host', 'external_url')
+        }),
+    )
+
+    @admin.display(description='Display Name')
+    def display_name(self, obj: Author):
+        return obj.display_name
+
+    @admin.display(description="Profile Image")
+    def profile_image_html(self, obj: Author):
+        if obj is None or obj.profile_image is None:
+            return format(f"<p>No image found</p>")
+
+        return format_html(f"<img src={obj.profile_image} width=100pt height=100pt />")
+
+
+    @admin.display(description='Location of Author')
+    def location(self, obj: Author):
+        return obj.location
+
+
+    def has_add_permission(self, request, obj=None):
+        return True
+
+    # TODO fix permissions for HTML button actions
+    def has_change_permission(self, request, obj: Author=None):
```

Comment:
does this need an issue?

Reply:
No, these are issues that affect certain admin panel buttons. It is not worth our time to fix it